File: backend/src/main/java/csc13001/plantpos/application/dtos/auth/LoginResponseDTO.java
package csc13001.plantpos.application.dtos.auth;

import csc13001.plantpos.domain.models.User;
import csc13001.plantpos.utils.http.JsonModel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import io.swagger.v3.oas.annotations.media.Schema;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class LoginResponseDTO extends JsonModel {

    @Schema(description = "User details after login")
    private User user;

    @Schema(example = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...")
    private String accessToken;
}

File: csc13001-plant-pos-frontend.Core/DTOs/Auth/LoginResponseDTO.cs
using csc13001_plant_pos_frontend.Core.Models;

namespace csc13001_plant_pos_frontend.Core.DTOs.Auth;

public class LoginResponseDTO
{
    public UserDTO User
    {
        get; set;
    }
    public string AccessToken
    {
        get; set;
    }
}

public class UserDTO
{
    public int UserId
    {
        get; set;
    }
    public string Username
    {
        get; set;
    }
    public string Role
    {
        get; set;
    } // API trả về "role"
    public bool Admin
    {
        get; set;
    } // API trả về "admin"
}

